karapace/schema_registry_apis.py

```diff
@@ -84,6 +84,22 @@ def __init__(self, config: Config) -> None:
 
         self._forward_client = None
         self.app.on_startup.append(self._create_forward_client)
+        self.health_hooks.append(self.schema_registry_health)
+
+    async def schema_registry_health(self) -> dict:
+        resp = {}
+        if self._auth is not None:
+            resp["schema_registry_authfile_timestamp"] = self._auth.authfile_last_modified
+        resp["schema_registry_ready"] = self.schema_registry.schema_reader.ready
+        resp["schema_registry_reader_offset"] = self.schema_registry.schema_reader.offset
+        resp["schema_registry_reader_highest_offset"] = self.schema_registry.schema_reader.highest_offset
+        cs = self.schema_registry.mc.get_coordinator_status()
+        resp["schema_registry_is_primary"] = cs.is_primary
+        resp["schema_registry_is_primary_eligible"] = cs.is_primary_eligible
+        resp["schema_registry_primary_url"] = cs.primary_url
+        resp["schema_registry_coordinator_running"] = cs.is_running
+        resp["schema_registry_coordinator_generation_id"] = cs.group_generation_id
+        return resp
```

Comment:
The highest offset is only read on startup so I'd make that explicit on the field name in JSON. See also other field name suggestions in the proposed format below.
And I would add also the current offset. And maybe add the highest offset from `OffsetWatcher`, the difference would be the lag and it should be small (0 or 1 in most cases). If the lag goes up then there is a problem.

Maybe format could be this?
```
{
  "schema_registry": {
    "authfile_modified_ts": self._auth.auth...,
    "ready": ..,
    "highest_offset_on_start": ..,
    "current_offset": ...,
    ....
  }
}
```

Reply:
I considered level structured response as there can be REST proxy related state variables too, but ended up with flat response for sake of simplicity.

Current code exposes current offset ie. the offset up to where schema reader has been reading messages. Did you meant to suggest about naming `schema_registry_reader_offset` -> `...current_offset` or did I miss something?

Incorporating `OffsetsWatcher`'s offset it is waiting into same exposed highest offset sounds like a plan to me.